scene/audio/audio_stream_player.cpp

```diff
@@ -62,6 +62,23 @@ void AudioStreamPlayer::_notification(int p_what) {
 			if (!playbacks_to_remove.is_empty()) {
 				emit_signal(SNAME("finished"));
 			}
+			if (!stream_playbacks.is_empty()) {
+				Ref<AudioStreamPlayback> playback = stream_playbacks[stream_playbacks.size() - 1];
+				int current_beat = playback->get_current_beat();
+				int current_bar = playback->get_current_bar();
+
+				if (current_bar >= 0) {
+					if (current_beat != prev_beat) {
+						emit_signal(SNAME("beat_changed"), current_beat);
+					}
+					if (current_bar != prev_bar) {
+						emit_signal(SNAME("bar_changed"), current_bar);
+					}
+
+					prev_beat = current_beat;
+					prev_bar = current_bar;
```

Comment:
```suggestion
					if (current_beat != prev_beat) {
						emit_signal(SNAME("beat_changed"), current_beat);
						prev_beat = current_beat;
					}
					if (current_bar != prev_bar) {
						emit_signal(SNAME("bar_changed"), current_bar);
						prev_bar = current_bar;
					}
```

see AudioStreamPlayer2D